project/sherLOCKED/Assets/Scripts/Objects/DoorInteraction.cs

```diff
@@ -20,22 +24,90 @@ void OnTriggerEnter2D(Collider2D other)
         Scene currentRoom = SceneManager.GetActiveScene ();
         // Get the name of this room
         string roomName = currentRoom.name;
+
+        // // Get the name of the door
+        // string doorName = 
+        
+        // Number of player objects in room
+        var numOfPlayers = GameObject.FindGameObjectsWithTag("Player").Length;
+        // for setting the player facing north
+        // facing = other.gameObject.GetComponent<Animator>();
+
         if (other.gameObject.CompareTag("Player") && !inCollider)
         {
+            Debug.Log( "collide (name) : " + GameObject.Find("dining_door") );
             inCollider = true;
-            DontDestroyOnLoad(other.gameObject);
-            //setting player postion after they go through a door            
-            playerPos = transform.position;
-            playerPos.x = -3;
-            playerPos.y = 2;
-            other.gameObject.transform.position = playerPos;
-            if(roomName == "Hall")
+
+            if(roomName == "Hall" && this.name == "library_door")
+            {
+                //setting player postion after they go through a door 
+                SceneManager.LoadScene("Library");           
```

Comment:
This is just an appearance thing, could we make sherry invisible (using `SetActive(false)`) while the new scene is being loaded and the player is being moved? Would just avoid the visual effect of Sherry teleporting around the room.

Reply:
Good tip!